src/serial.rs

```diff
@@ -306,11 +307,11 @@ macro_rules! hal {
                     buffer: B,
                 ) -> Transfer<R, B, $tx_chan, Self>
                 where
-                    A: Unsize<[u8]>,
+                    A: AsRef<[u8]>,
                     B: Static<A>,
                 {
                     {
-                        let buffer: &[u8] = buffer.borrow();
+                        let buffer = buffer.borrow().as_ref();
```

Comment:
I don't like that, mutability seems wrong. I think it must be `buffer.borrow_mut().as_mut()`, but I would need to add `DerefMut` to `Static`, which can't be safe. As proposed, that's as before.